data_collection/gazette/spiders/sp_sao_paulo.py

```diff
@@ -0,0 +1,54 @@
+import locale
+import re
+from datetime import date, datetime, timedelta
+
+import scrapy
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+RE_max_page_num = re.compile(r"\d+ de (\d+)")
+
+
+class SpSaoPauloSpider(BaseGazetteSpider):
+    TERRITORY_ID = "3550308"
+    BASE_URL = "http://diariooficial.imprensaoficial.com.br"
+    allowed_domains = ["diariooficial.imprensaoficial.com.br"]
+    name = "sp_sao_paulo"
+    start_date = date(2017, 6, 1)
+
+    def start_requests(self):
+        # Need to have the month's name in portuguese for the pdf url
+        locale.setlocale(locale.LC_TIME, "pt_BR.UTF-8")
+        today = date.today()
+        day = self.start_date
+        while day < today:
+            url = f"{self.BASE_URL}/nav_v6/header.asp?txtData={day.strftime('%d/%m/%Y')}&cad=1"
+            yield scrapy.Request(url, cb_kwargs=dict(day=day))
+            day = day + timedelta(days=1)
```

Comment:
This is not wrong. But I think we can share other approaches as well. Other spiders use `rrule` to generate a date range. If you are interested, take a look. ;)

Reply:
Oh, indeed that's nicer. I'll make the change to use it shortly.